mediapipe/tasks/c/vision/pose_landmarker/pose_landmarker_result_converter.cc

```diff
@@ -0,0 +1,97 @@
+
+namespace mediapipe::tasks::c::components::containers {
+
+using CppLandmark = ::mediapipe::tasks::components::containers::Landmark;
+using CppNormalizedLandmark =
+    ::mediapipe::tasks::components::containers::NormalizedLandmark;
+
+void CppConvertToPoseLandmarkerResult(
+    const mediapipe::tasks::vision::pose_landmarker::PoseLandmarkerResult& in,
+    PoseLandmarkerResult* out) {
+  if (in.segmentation_masks.has_value()) {
+    out->segmentation_masks_count = in.segmentation_masks.value().size();
+    out->segmentation_masks = new MpImage[out->segmentation_masks_count];
+    for (uint32_t i = 0; i < out->segmentation_masks_count; ++i) {
+      const auto& image_frame =
+          in.segmentation_masks.value()[i].GetImageFrameSharedPtr();
+      out->segmentation_masks[i] = {
+          .type = MpImage::IMAGE_FRAME,
+          .image_frame = {
+              .format = static_cast<::ImageFormat>(image_frame->Format()),
+              .image_buffer = image_frame->PixelData(),
+              .width = image_frame->Width(),
+              .height = image_frame->Height()}};
+    }
+  } else {
+      out->segmentation_masks_count = 0;
+      out->segmentation_masks = nullptr;
+  }
+
+  out->pose_landmarks_count = in.pose_landmarks.size();
+  out->pose_landmarks = new NormalizedLandmarks[out->pose_landmarks_count];
+  for (uint32_t i = 0; i < out->pose_landmarks_count; ++i) {
+    std::vector<CppNormalizedLandmark> cpp_normalized_landmarks;
+    for (uint32_t j = 0; j < in.pose_landmarks[i].landmarks.size(); ++j) {
+      const auto& cpp_landmark = in.pose_landmarks[i].landmarks[j];
+      cpp_normalized_landmarks.push_back(cpp_landmark);
+    }
+    CppConvertToNormalizedLandmarks(cpp_normalized_landmarks,
+                                    &out->pose_landmarks[i]);
+  }
+
+  out->pose_world_landmarks_count = in.pose_world_landmarks.size();
+  out->pose_world_landmarks = new Landmarks[out->pose_world_landmarks_count];
+  for (uint32_t i = 0; i < out->pose_world_landmarks_count; ++i) {
+    std::vector<CppLandmark> cpp_landmarks;
+    for (uint32_t j = 0; j < in.pose_world_landmarks[i].landmarks.size(); ++j) {
+      const auto& cpp_landmark = in.pose_world_landmarks[i].landmarks[j];
+      cpp_landmarks.push_back(cpp_landmark);
+    }
+    CppConvertToLandmarks(cpp_landmarks, &out->pose_world_landmarks[i]);
+  }
+}
+
+void CppClosePoseLandmarkerResult(PoseLandmarkerResult* result) {
+  for (uint32_t i = 0; i < result->pose_landmarks_count; ++i) {
+    CppCloseNormalizedLandmarks(&result->pose_landmarks[i]);
+  }
+  delete[] result->pose_landmarks;
+
+  for (uint32_t i = 0; i < result->pose_world_landmarks_count; ++i) {
+    CppCloseLandmarks(&result->pose_world_landmarks[i]);
+  }
+  delete[] result->pose_world_landmarks;
+
+  result->pose_landmarks = nullptr;
+  result->pose_world_landmarks = nullptr;
+
+  result->pose_landmarks_count = 0;
+  result->pose_world_landmarks_count = 0;
```

Comment:
Need to free the masks here as well.

Reply:
Since segmentation_masks[i].image_frame.image_buffer is not dynamically allocated here, we do not delete it in the cleanup, right?

Reply:
I think you still need to clear https://github.com/google/mediapipe/pull/5052/files/cc0d3466afaba965520ae9ac2578eefb2da3e5c9#diff-497245f04089ed54cca2729d05cf2d0be6c370d1d2072d5c4b0a606345cb22bdR38 ?

Reply:
@schmidt-sebastian I've added it now, but I get a SIGSEGV when I try to use free or delete on `const uint8_t* image_buffer`. Could you take a look? I believe casting away its constness and deleting it can lead to undefined behavior

Reply:
I'll debug during import.

Reply:
You actually only need to free the image array and not the data itself since the data is statically allocated (which could present a problem depending on how the API is used, but this is outside of the scope of this PR).